src/main/scala/scalaz/stream/Process.scala

```diff
@@ -871,6 +871,24 @@ object Process extends ProcessInstances {
   def iterateEval[F[_], A](start: A)(f: A => F[A]): Process[F, A] =
     emit(start) ++ await(f(start))(iterateEval(_)(f))
 
+  /**
+   * Create an iterator to use as a source for a `Process`,
+   * which lazily emits the values of the iterator, then halts.
+   *
+   * Be sure that iteratorCreator creates any mutable or external
+   * resources that the iterator uses are created as part of the
+   * Task. Also, do not use Task.now to create the iterator.
+   */
+  def iterator[F[_], O](iteratorCreator: F[Iterator[O]]): Process[F, O] = {
```

Comment:
I like the effectful `iteratorCreator` concept.  I initially looked at this signature with some suspicion, but now I believe it is the correct way to go (modulo my concerns below).